experimenter/experimenter/nimbus_ui_new/static/js/edit_branches.js

```diff
@@ -0,0 +1,50 @@
+import { basicSetup } from "codemirror";
+import { EditorView } from "@codemirror/view";
+import { json, jsonParseLinter } from "@codemirror/lang-json";
+import { linter } from "@codemirror/lint";
+import { autocompletion } from "@codemirror/autocomplete";
+import { schemaAutocomplete, schemaLinter } from "./validator.js";
+import $ from "jquery";
+
+const setupCodemirror = () => {
+  const textareas = document.querySelectorAll(".value-editor");
+
+  textareas.forEach((textarea) => {
+    const jsonSchema = JSON.parse(textarea.dataset["schema"]);
+
+    const extensions = [
+      basicSetup,
+      EditorView.updateListener.of((v) => {
+        if (v.docChanged) {
+          const value = v.state.doc.toString();
+          const textarea = v.view.dom.parentNode.querySelector(".value-editor");
+          textarea.value = value;
+        }
+      }),
+      json(),
+      linter(jsonParseLinter()),
+      linter(schemaLinter(jsonSchema)),
+      autocompletion({ override: [schemaAutocomplete(jsonSchema)] }),
+    ];
+
+    let view = new EditorView({
+      doc: textarea.value || "{}",
+      extensions,
+      parent: textarea.parentNode,
+    });
+
+    view.dom.style.border = "1px solid #ccc";
```

Comment:
This seems like something we could put in a CSS file :P

Reply:
🙈